migrate.py

```diff
@@ -649,6 +690,314 @@ def copy_tests(plugin, coll, spec, args):
                         with open(yf, 'w') as f:
                             f.write(ydata)
 
+##############################################################################
+# Rewrite integration tests
+##############################################################################
+
+integration_tests_deps = set()
+
+def integration_tests_add_to_deps(collection, dep_collection):
```

Comment:
Put a guard expression here. Doing this check before *each* invocation makes the code unnecessarily nested which reduces readability.

```suggestion
def integration_tests_add_to_deps(collection, dep_collection):
    if collection == dep_collection:
        return
```